release-common-lib/src/main/java/org/reactome/util/compare/DBObjectComparer.java

```diff
@@ -147,225 +176,428 @@ public static int compareInstances(
-							{
-								if (values1.size() == values2.size())
-								{
-									for (int i = 0 ; i < values1.size() ; i++)
-									{
-										// compare each item in one list to the corresponding item in the other list -
-										// the MySQLAdaptor seems to preserve sequence of items in lists properly.
-										if (!values1.get(i).equals(values2.get(i)))
-										{
-											sb.append(
-												indentString + "Mismatch on" + attributeQualifier + " attribute \"" +
-												attrib.getName() + "\"\n" + indentString + "Instance 1 (\"" +
-												inst1.toString() + "\") has value:\t" + values1.get(i) + "\n" +
-												indentString + "Instance 2 (\"" + inst2.toString() + "\") has value:" +
-												"\t" + values2.get(i)
-											).append("\n");
-											count++;
-										}
-									}
-								}
-								else
-								{
-									// no point comparing if the lists don't even have the same size. Just write a
-									// message about that.
-									sb.append(
-										indentString + "Count mismatch for multi-valued" + attributeQualifier +
-										" attribute \"" + attrib.getName() + "\" Instance 1 (\"" + inst1.toString() +
-										"\") has " + values1.size() + " elements but Instance 2 (\""+inst2.toString() +
-										"\") has " + values2.size() + " elements.\n"
-									);
-									count ++;
-								}
-							}
-						}
-					}
+					count = compareValuesOfAttributeBetweenInstances(
+						attribute, value1, value2, instance1, instance2, stringBuilder, count, recursionDepth,
+						maxRecursionDepth, customAttributeNameFilter, checkReferrers
+					);
 				}
 			}
+			else
+			{
+				stringBuilder.append(getIndentString(recursionDepth))
+					.append(getCountMismatchMessage(instance1, instance2, attribute));
+
+				count++;
+			}
 		}
 		return count;
 	}
+
+	/**
+	 * Checks passed GKInstance object to determine if it is an InstanceEdit instance or not.  Returns
+	 * <code>true</code> if the instance is an InstanceEdit; <code>false</code> otherwise
+	 * @param instance GKInstance object to check
+	 * @return <code>true</code> if instance is an InstanceEdit; <code>false</code> otherwise
+	 */
+	private static boolean isInstanceEdit(GKInstance instance)
+	{
+		return instance.getSchemClass().getName().equals("InstanceEdit");
+	}
+
+	/**
+	 * Compares two instances and determines if their schema class types are different.  Returns true if the instances
+	 * are of different types; false otherwise
+	 * @param instance1 First instance to compare
+	 * @param instance2 Second instance to compare
+	 * @return <code>true</code> if the instances are of different types; <code>false</code> otherwise
+	 */
+	private static boolean differentInstanceTypes(GKInstance instance1, GKInstance instance2)
+	{
+		String instanceType1 = instance1.getSchemClass().getName();
+		String instanceType2 = instance2.getSchemClass().getName();
+
+		return !instanceType1.equals(instanceType2);
```

Comment:
For _brevity's_ sake, it could be 
`return !instanceType1.getSchemClass().getName().equals(instance2.getSchemClass().getName())`,
though I can see that might not be the best for _clarity's_ sake

Reply:
There's always a trade-off, and when that happens, you'll always get dissenting opinions. Fun times... ;)

Reply:
Yeah, I considered implementing it in one line, but I thought the extra lines were worth it for making it clear that the types of the instances were being obtained and compared.